Rename package add command to hcl module and document runtime loading (#20709)
pulumi/docs: content/docs/iac/get-started/terraform/terraform-modules.md

## Leverage the module ecosystem

Pulumi can directly use existing Terraform modules from the Terraform Registry, private registries, or local sources. This allows you to access thousands of existing modules without rewriting them in Pulumi.
It's powered by the [`hcl` provider](/registry/packages/hcl/), which turns any Terraform or OpenTofu module into a Pulumi component, either as a strongly typed SDK or loaded dynamically at runtime.

## Add Terraform modules

Use the `pulumi package add` command to add Terraform modules to your project:

```bash
# Add a module from the Terraform Registry

> $ pulumi package add hcl module terraform-aws-modules/vpc/aws 6.0.0 vpc

Reviewer: ```suggestion
$ pulumi package add hcl module terraform-aws-modules/vpc/aws 6.0.0
```

# Add a local module

> $ pulumi package add hcl module ./path/to/module localmod

Reviewer: ```suggestion
$ pulumi package add hcl module ./path/to/module
```

Next, add the VPC module:

```bash

> $ pulumi package add hcl module terraform-aws-modules/vpc/aws 6.0.0 vpc

Reviewer: ```suggestion
$ pulumi package add hcl module terraform-aws-modules/vpc/aws 6.0.0
```

{{% /choosable %}}

## Load a module at runtime

> Generating an SDK gives you strongly typed inputs and outputs, IDE completion, and a package you can pin and share across your team. When you'd rather trade that type safety for flexibility — for example, to load a module whose address isn't known until runtime — you can use the [`hcl` provider](/registry/packages/hcl/) instead of generating an SDK.

Author: Feel free to change the name of this if it should be called something else (the "HCL Module provider", maybe?).

> Generating an SDK gives you strongly typed inputs and outputs, IDE completion, and a package you can pin and share across your team. When you'd rather trade that type safety for flexibility — for example, to load a module whose address isn't known until runtime — you can use the [`hcl` provider](/registry/packages/hcl/) instead of generating an SDK.

Reviewer: The strongly typed SDK is also using the `hcl` provider. It's the same thing with 2 access stories: static & dynamic. We don't have to explain it, since the user doesn't need to know, but our docs should be correct at the technical level.

```suggestion
Generating an SDK gives you strongly typed inputs and outputs, IDE completion, and a package you can pin and share across your team. When you'd rather trade that type safety for flexibility — for example, to load a module whose address isn't known until runtime — you can use the [`hcl` provider](/registry/packages/hcl/) directly.
```
